bodhi/server/models/models.py

```diff
@@ -789,7 +789,10 @@ def edit(cls, request, data):
 
                 # Expire any associated buildroot override
                 if b.override:
-                    b.override.expire()
+                    # Untag buildroot override only if the update reaches stable state or
+                    # the update reaches expired date
+                    if up.status == UpdateStatus.stable or datetime.utcnow() >= b.override.expired_date:
+                        b.override.expire()
```

Comment:
I'm not confident that this code is why #798 is happening. Can you explain a bit why you think this is the particular call to expire that is causing the issue? It sounds like the reporter experiences this issue when the mash happens and not when editing an update, but I admit that it is possible that the problem is here in this code. Can you explain a little more since you've clearly thought about it more than I have?